### DISCUSSION TITLE
mujoco Collision problems

### DISCUSSION THREAD

**Discussion by 2802029572** [google-deepmind/mujoco#2283] [Asking for Help]
### Intro

Hi!

I am a graduate student , I'm using mujoco for robotic arm obstacle avoidance reinforcement learning training, and during the training, the robotic arm bounces off when it hits an obstacle. Why is this and how can I fix it?

### My setup

mujoco

### My question

I am a graduate student , I'm using mujoco for robotic arm obstacle avoidance reinforcement learning training, and during the training, the robotic arm bounces off when it hits an obstacle. Why is this and how can I fix it?

### Minimal model and/or code that explain my question

<body name="base_link" pos="-0.5 -0.1 0.43">
                <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="4" diaginertia="0.0072 0.00443333 0.00443333" />
                <geom name= "base_link" type="mesh"  mesh="base" material="ur5_mat" rgba="0.5 0.5 0.5 1" />
                <body name="shoulder_link" pos="0 0 0.089159">
                    <inertial pos="0 0 0" mass="3.7" diaginertia="0.0102675 0.0102675 0.00666" />
                    <joint name="shoulder_pan_joint" class="UR5" pos="0 0 0" axis="0 0 1" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/>
                    <geom name="shoulder_link" type="mesh"  mesh="shoulder" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                    <body name="upper_arm_link" pos="0 0.13585 0" >
                        <inertial pos="0 0 0.28" mass="8.393" diaginertia="0.226891 0.226891 0.0151074" />
                        <joint name="shoulder_lift_joint" class="UR5" pos="0 0 0" axis="0 1 0" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/> 
                        <geom name="upper_arm_link" type="mesh"  mesh="upperarm" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                        <body name="forearm_link" pos="0 -0.1197 0.425">
                            <inertial pos="0 0 0.25" mass="2.275" diaginertia="0.0494433 0.0494433 0.004095" />
                            <joint name="elbow_joint" class="UR5" pos="0 0 0" axis="0 1 0" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/>
                            <geom name="elbow_joint" type="mesh"  mesh="forearm" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                            <body name="wrist_1_link" pos="0 0 0.39225" quat="0.707107 0 0.707107 0">
                                <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="1.219" diaginertia="0.21942 0.111173 0.111173" />
                                <joint name="wrist_1_joint" class="UR5e" pos="0 0 0" axis="0 1 0" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/>
                                <geom name="wrist_1_link" type="mesh"  mesh="wrist1" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                                <body name="wrist_2_link" pos="0 0.093 0">
                                    <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="1.219" diaginertia="0.21942 0.111173 0.111173" />
                                    <joint name="wrist_2_joint" class="UR5e" pos="0 0 0" axis="0 0 1" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/>
                                    <geom name="wrist_2_link" type="mesh"  mesh="wrist2" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                                    <body name="wrist_3_link" pos="0 0 0.09465">
                                        <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="0.1879" diaginertia="0.033822 0.0171365 0.0171365" />
                                        <joint name="wrist_3_joint" class="UR5e" pos="0 0 0" axis="0 1 0" limited="true" range="-3.14159 3.14159" damping="300"  armature="1" stiffness="0"/>
                                        <geom name="wrist_3_link" type="mesh"  mesh="wrist3" material="ur5_mat" rgba="0.5 0.5 0.5 1"/>
                                        <body name="ee_link" pos="0 0.0823 0" quat="0.707107 0 0 0.707107">
                                            <inertial pos="-0.01 0 0" mass="0.001" diaginertia="1.66667e-08 1.66667e-08 1.66667e-08" />
                                            <geom size="0.005 0.005 0.005" pos="-0.01 0 0" type="box" rgba="0.5 0.5 0.5 1"/>
                                            <site name="ee_site" pos="0.15 0 0" type = "sphere"  size = "0.02" rgba = "1 0 0 1 "   group = "1"/>

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by AumGupta** [comment]
Hey @2802029572 ,
Found something similar
https://forum.coppeliarobotics.com/viewtopic.php?p=41441&sid=74ecadcf368044f6f57d8a4d0a5d5304#p41441

**Comment by btaba** [comment]
Hi @2802029572 it sounds like you'll want to adjust the stiffness/damping of your contact, via [`solref`](https://mujoco.readthedocs.io/en/stable/modeling.html#reference) on the collision geoms.
